carbon-design-system/carbon-addons-iot-react
# 3100 list inconsistent checkbox behaviour

Closes #3100

**Summary**
It was hard to tell why this bug occurred but it turned out it was related to the carbon checkbox bug that we thought we already had a workaround for. 

**Change List (commits, features, bugs, etc)**
The checkbox emits a double click event, which will trigger double click events for our wrapping list rows. But, the actual workaround is simple, we need to stop propagation on the checkbox click even though we weren't using it. 

**Acceptance Test (how to verify the PR)**
1. Go to https://deploy-preview-3105--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-list--with-checkbox-multi-selection
2. Click on a list item's checkbox and notice that the Item is selected
3. Check the isVirtual knob
4. Click on another list item's checkbox and notice that the Item is also selected

**Regression Test (how to make sure this PR doesn't break old functionality)**
Verify that the other List and HierarchyList stories that use checkboxes works as expected. For the hierarchyList you can use this story knob config
https://deploy-preview-3105--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-hierarchylist--with-nested-reorder&knob-Allow%20Item%20Movement=true&knob-Title=MLB%20Expanded%20List&knob-Default%20Selected%20Id=New%20York%20Mets_Pete%20Alonso&knob-Editing%20Style=multiple-nesting&knob-Page%20Size=lg&knob-isLoading=&knob-isLargeRow=&knob-hasSearch=true&knob-hasDeselection=true&knob-hasMultiSelect=&knob-hasVirtualList=&knob-lockedIds=["New%20York%20Mets_Jeff%20McNeil"]



**Things to look for during review**
